### PR TITLE
feat: add foundation effects for `java.net`

### PR DESCRIPTION
**Pull request by AStenbaek**
I have manually gone through the `java.net` library in openjdk 21 and added base effects to associated classes and methods. The majority of addition are whole classes being associated with the `Net` effect. The `URLClassLoader` class also has the `Sys` effect since it can load a class.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@AStenbaek You need to run the tests...

**Comment by AStenbaek** [comment]
Fixed the issue with compilation.
Test suite fails two tests:  [Test.Exp.Jvm.Complex.flix](https://github.com/flix/flix/blob/master/main/test/flix/Test.Exp.Jvm.InvokeMethod.flix) and [Test.Exp.Jvm.InvokeMethod.flix](https://github.com/flix/flix/blob/master/main/test/flix/Test.Exp.Jvm.InvokeMethod.flix).

I assume you want me to adjust the expected effects?

File: main/src/ca/uwaterloo/flix/language/phase/typer/BaseEffects.Classes.json [hunk 1]
```diff
@@ -3,6 +3,22 @@
...
+    "java.net.NetMulticastSocket": "Net",
+    "java.net.NetworkInterface": "Net",
+    "java.net.ProxySelector": "Net",
+    "java.net.ResponseCache": "Net",
+    "java.net.URL": "Net",
+    "java.net.URLClassLoader": "Net, Sys",
+    "java.net.URLConnection": "Net",
+    "java.net.URLStreamHandler": "Net",
+    "java.net.spi.InetAddressResolverProvider": "Net",
+    "java.net.spi.URLStreamHandlerProvider": "Net"
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/typer/BaseEffects.Classes.json:22]: Lets keep the list sorted.

File: main/src/ca/uwaterloo/flix/language/phase/typer/BaseEffects.Methods.json [hunk 2]
```diff
@@ -1,6 +1,10 @@
 {
   "methods": {
     "java.lang.Runtime::exec": "Exec",
-    "java.lang.System::exit": "Sys"
+    "java.lang.System::exit": "Sys",
+    "java.net.CookieHandler::getDefault": "Net",
+    "java.net.CookieHandler::setDefault": "Net",
+    "java.net.InetAddress::getByName": "Net",
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/typer/BaseEffects.Methods.json:7]: Why not tag all of InetAddress with Net?

File: main/src/ca/uwaterloo/flix/language/phase/typer/BaseEffects.Classes.json (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/typer/BaseEffects.Classes.json:22]: Also, we need to add the `IO` effect to everything, so lets make it `[Net, IO]`.

**Comment by AStenbaek** [comment]
My TODO:
- [ ] Add `IO` to everything.
- [ ] Sort content of json files.
- [ ] Take another look at the net package to see if I missed anything.

File: main/src/ca/uwaterloo/flix/language/phase/Safety.scala [hunk 3]
```diff
@@ -216,13 +216,17 @@ object Safety {
...
-      case Type.Sys => true
-      case _ => false
+    def isAllowedEffect0(eff: Type): Boolean = {
+      eff match {
+        case Type.Pure => true
+        case Type.IO => true
+        case Type.Net => true
+        case Type.NonDet => true
+        case Type.Sys => true
+        case ts => ts.typeVars.forall(isAllowedEffect0)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Safety.scala:226]: A type var is never an effect?

**Comment by magnus-madsen** [comment]
@AStenbaek What's the status on this?

File: main/src/ca/uwaterloo/flix/language/phase/Safety.scala (hunk 3 above)
Comment (AStenbaek) [reply main/src/ca/uwaterloo/flix/language/phase/Safety.scala]: I missed the `effects` method from the `Type` type. This should be fixed now.

File: main/src/ca/uwaterloo/flix/language/phase/Safety.scala [hunk 4]
```diff
@@ -216,11 +216,15 @@ object Safety {
     * Returns `true` if the given `defn` is pure or has an effect that is allowed for a top-level function.
     */
   private def isAllowedEffect(defn: Def): Boolean = {
-    defn.spec.eff match {
-      case Type.Pure => true
-      case Type.IO => true
-      case Type.NonDet => true
-      case Type.Sys => true
+    defn.spec.eff.effects.iterator forall {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Safety.scala:219]: Drop iterator. add dot before forAll

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Merged. Thanks!

**Pull request merged**